odbc/src/api/get_info_tests.rs

```diff
@@ -0,0 +1,633 @@
+                $(buffer_length = $buffer_length;)?
+
+                // Assert that the actual result matches expected
+                assert_eq!(
+                    $expected_sql_return,
+                    SQLGetInfoW(
+                        mongo_handle as *mut _,
+                        info_type,
+                        value_ptr as Pointer,
+                        buffer_length,
+                        out_length,
+                    )
+                );
+
+                // If the expectation is that the function returns successfully,
+                // assert that the resulting value and length are correct
+                match $expected_sql_return {
+                    SqlReturn::SUCCESS => {
+                        $(assert_eq!($expected_length, *out_length);)?
+                        $(assert_eq!($expected_value, $actual_value_modifier(value_ptr, *out_length as usize));)?
+                    },
+                    _ => ()
+                }
+
+                let _ = Box::from_raw(value_ptr);
+            }
+        }
+    }
+}
+
+unsafe fn modify_string_value(value_ptr: Pointer, out_length: usize) -> String {
+    input_wtext_to_string(value_ptr as *const _, out_length)
+}
+
+unsafe fn modify_u32_value(value_ptr: Pointer, _: usize) -> u32 {
+    *(value_ptr as *mut UInteger)
+}
+
+unsafe fn modify_u16_value(value_ptr: Pointer, _: usize) -> u16 {
+    *(value_ptr as *mut USmallInt)
+}
+
+mod unit {
+    use super::*;
+
+    test_get_info!(
+        driver_name,
+        info_type = InfoType::DriverName,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 40,
+        expected_length = 39,
+        expected_value = "MongoDB Atlas SQL interface ODBC Driver",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        driver_ver,
+        info_type = InfoType::DriverVer,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 11,
+        expected_length = 10,
+        expected_value = "00.01.0000",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        driver_odbc_ver,
+        info_type = InfoType::DriverOdbcVer,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 6,
+        expected_length = 5,
+        expected_value = "03.08",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        search_pattern_escape,
+        info_type = InfoType::SearchPatternEscape,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 1,
+        expected_length = 0,
+        expected_value = "",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        dbms_name,
+        info_type = InfoType::DbmsName,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 14,
+        expected_length = 13,
+        expected_value = "MongoDB Atlas",
```

Comment:
This one also for the `constants` package.

Reply:
done